src/app/users/user.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject } from "rxjs/internal/BehaviorSubject";
+import { User, UserRole } from "./user.model";
+import { first, map, Observable, tap } from "rxjs";
+import { HttpClient } from "@angular/common/http";
+import { beApiRoutes } from "../routes/be-api.routes";
+import { Option } from '../shared/option.model';
+
+@Injectable({ providedIn: "root" })
+export class UserService{
+    
+    public usersSubject = new BehaviorSubject<User[]>([]);
+    public usersObservable$ = this.usersSubject.asObservable();
```

Comment:
you can remove the Observable name, just users$ the $ sign signifies that it's an observable